packages/gatsby/src/redux/actions/public.ts

```diff
@@ -1154,12 +1189,16 @@ actions.setBabelOptions = (options: Object, plugin?: ?Plugin = null) => {
  *   },
  * })
  */
-actions.setBabelPlugin = (config: Object, plugin?: ?Plugin = null) => {
+actions.setBabelPlugin = (
+  config: Record<string, any>,
+  plugin: IGatsbyPlugin | null = null
```

Comment:
again, I would change this to not handle the `null` case as we weren't handling it before.

Reply:
Done in 3ccf334